src/hooks/useOnClickOutside.tsx

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+
+export interface UseRefObj {
+  current: null | HTMLDivElement;
+}
+
+const useOnClickOutside = (ref: UseRefObj, outsideClickCallback: Function) => {
+  useEffect(() => {
+    function handleClickOutside(event: Event) {
```

Comment:
Could you change this to arrow notation just for consistency? `const a = () => {}`